kernel/src/vm/mod.rs

```diff
@@ -49,3 +65,221 @@ pub fn mem_total() -> usize {
 
     total
 }
+
+/// On construction, PauseProcGaurd will stop the process it holds a reference to, and when dropped
+/// will resume the process. It is safe to resume the process before the drop.
+struct PauseProcGaurd {
+    proc: Arc<Process>,
+}
+
+impl PauseProcGaurd {
+    fn new(proc: Arc<Process>) -> Self {
+        // TODO(aneesh): Can this be done at the scheduler level instead?
+        proc.stop(SIGSTOP);
```

Comment:
Thread::park?

Reply:
As discussed, just use SIGSTOP for now. We'll figure it out later. We definitely need to be able to control things better.